umm-lib/src/cmr/umm/iso_smap/granule/orbit.clj

```diff
@@ -50,66 +69,170 @@
             start-lat (convert-direction start-direction)
             end-lat (convert-direction end-direction))))
 
+(defn- build-ocsd-string
+  "Builds ISO SMAP orbit calculated spatial domain string using umm values."
+  [ocsd]
+  (let [{:keys [orbital-model-name orbit-number start-orbit-number stop-orbit-number
+                equator-crossing-longitude equator-crossing-date-time]} ocsd
+        OrbitalModelName (when orbital-model-name
+                           (str "OrbitalModelName: " orbital-model-name))
+        OrbitNumber (when orbit-number
+                      (str "OrbitNumber: " orbit-number))
+        StartOrbitNumber (when start-orbit-number
+                           (str "BeginOrbitNumber: " start-orbit-number))
+        StopOrbitNumber (when stop-orbit-number
+                          (str "EndOrbitNumber: " stop-orbit-number))
+        EquatorCrossingLongitude (when equator-crossing-longitude
+                                   (str "EquatorCrossingLongitude: " equator-crossing-longitude))
+        EquatorCrossingDateTime (when equator-crossing-date-time
+                                  (str "EquatorCrossingDateTime: " equator-crossing-date-time))]
+    (str OrbitalModelName " " OrbitNumber " " StartOrbitNumber " " StopOrbitNumber " "
+         EquatorCrossingLongitude " " EquatorCrossingDateTime)))
+
 (defn- parse-float
   "Coerce's string to float, catches exceptions and logs error message and returns nil if
   value is not parseable."
-  [field value]
+  [field value] 
   (try
     (Float. value)
     (catch Exception e
-      (info (format "For orbit field [%s] the value [%s] is not a number." field value))
-      ;; We return nil her instead of value because within-range validation can't handle comparing a
+      (info (format "For Orbit field [%s] the value [%s] is not a number." field value))
+      ;; We return nil here instead of value because within-range validation can't handle comparing a
+      ;; string to a double.
+      nil)))
+
+(defn- parse-integer
+  "Coerce's string to integer, catches exceptions and logs error message and returns value if
+  value is not parseable so that the validation can catch the error and return to the user."
+  [field value]
+  (try
+    (Integer. value)
+    (catch Exception e
+      (info (format "For Orbit calculated spatial domain field [%s] the value [%s] is not an integer." field value))
+      value)))
+
+(defn- parse-datetime
+  "Coerce's string to datetime, catches exceptions and logs error message and returns value if
+  value is not parseable. This wrapper is used so that the error can be returned together 
+  with other validation errors." 
+  [field value]
+  (try
+    (date-time-parser/parse-datetime value)
+    (catch Exception e
+      (format "For Orbit calculated spatial domain field [%s] the value [%s] is not a datetime." field value)
```

Comment:
This line doesn't do anything - I assume we want to log a message.

Reply:
Yeah, it's supposed to be (info ... ). Done.